coffee_httpd/src/httpd/server.rs

```diff
@@ -178,6 +179,27 @@ async fn coffee_remote_list(data: web::Data<AppState>) -> Result<Json<Value>, Er
     }
 }
 
+#[api_v2_operation]
+#[get("/show")]
+async fn coffee_show(data: web::Data<AppState>, body: Json<Show>) -> Result<Json<Value>, Error> {
+    let plugin = &body.plugin;
+
+    let mut coffee = data.coffee.lock().await;
+    let result = coffee.show(plugin).await;
+
+    match result {
+        Ok(coffee_show) => {
+            let val = serde_json::to_value(coffee_show).map_err(|err| {
+                actix_web::error::ErrorInternalServerError(format!("Failure: {err}"))
+            })?;
+            Ok(Json(val))
+        }
+        Err(err) => Err(actix_web::error::ErrorInternalServerError(format!(
+            "Failure: {err}"
+        ))),
+    }
```

Comment:
The `handle_httpd_response` macro handles responses where there is a success message. 
It doesn't handle `list`, `remote/list` or `show` because these have a returned value that gets parsed in case the result is `ok`.

I defined another macro to handle responses for these endpoints

Reply:
I do not think you need another macro. But you need just a macro that accepts the result content

Reply:
Ok I changed it to be 1 macro with 2 different conditions:
1- there is no return value in the result but we want to return a success message
2- there is a return value

Now we use the same macro in every endpoint in the server